website/static/js/share/sideBar.js

```diff
@@ -0,0 +1,85 @@
+var m = require('mithril');
+var $osf = require('osfHelpers');
+var utils = require('./utils.js');
+
+var SideBar = {};
+
+SideBar.view = function(ctrl){
+    if (ctrl.vm.results === null){
+        return [];
+    }
+    return m('.row', [
+        m('.btn-group', [
+            m('button.btn.btn-default.dropdown-taggle', {
+                    'data-toggle': 'dropdown',
+                    'aria-expanded': 'false'
+                }, ['Sort by: ' + ctrl.vm.sort(), m('span.caret')]
+            ),
+            m('ul.dropdown-menu', {'role': 'menu'}, [
+                m('li', [m('a', {
+                    'href': '#',
+                    onclick: function(event) {
+                            ctrl.vm.sort('Relevance');
+                            utils.search(ctrl.vm);
+                        }
+                }, ['Relevance'])]),
+                m('li', [m('a', {
+                    'href': '#',
+                    onclick: function(event) {
+                        ctrl.vm.sort('Date');
+                        utils.search(ctrl.vm);
+                    }
+                }, ['Date'])]),
+            ])
+        ]),
+        m('br'),
+        m('ul', {style:{'list-style-type': 'none'}}, ctrl.renderProviders()),
+    ]);
+};
+
+SideBar.controller = function(vm) {
+    var self = this;
+    self.vm = vm;
+
+    self.vm.sort = m.prop($osf.urlParams().sort) || m.prop("Relevance");
+    self.vm.requiredFilters = $osf.urlParams().required || [];
```

Comment:
I do split them below, but if you do this it throws an exception if they are not defined.